lib/Sema/TypeOfReference.cpp

```diff
@@ -849,19 +849,27 @@ FunctionType *ConstraintSystem::adjustFunctionTypeForConcurrency(
       } else if (numApplies < decl->getNumCurryLevels() &&
                  decl->hasCurriedSelf() ) {
         auto shouldMarkMemberTypeSendable = [&]() {
-          // Static member types are @Sendable on both levels because
-          // they only capture a metatype "base" that is always Sendable.
-          // For example, `(S.Type) -> () -> Void`.
-          if (!decl->isInstanceMember())
-            return true;
+          Type capturedBaseType = baseType;
+
+          if (!decl->isInstanceMember()) {
```

Comment:
https://github.com/swiftlang/swift/blob/main/lib/Sema/CSSimplify.cpp#L10575-L10576 has to be adjusted to match this to make sure that members are delayed if there are any type variables involved in the base type. Please add a test-case for that, we should have had one that would crash in this case without change in `performMemberLookup`...

Reply:
Thank you! I had complete missed this. Fix + test case added: it only needs a narrow carve-out for members of protocols, because all concrete types have sendable metatypes.

Reply:
Great, thank you!